src/recipe.cpp

```diff
@@ -418,52 +418,49 @@ bool recipe::has_byproducts() const
     return !byproducts.empty();
 }
 
-std::string recipe::required_skills_string( const Character *c, bool print_skill_level ) const
+std::string recipe::primary_skill_string( const Character *c, bool print_skill_level ) const
 {
-    if( required_skills.empty() ) {
-        if( difficulty == 0 ) {
-            return _( "<color_cyan>none</color>" );
-        } else {
-            const int player_skill = c ? c->get_skill_level( skill_used ) : 0;
-            std::string difficulty_color = difficulty > player_skill ? "yellow" : "green";
-            std::string skill_level_string = print_skill_level ? "" :
-                                             ( std::to_string( player_skill ) + "/" );
-            skill_level_string += std::to_string( difficulty );
-            return string_format( "<color_cyan>%s</color> <color_%s>(%s)</color>",
-                                  skill_used.obj().name(), difficulty_color, skill_level_string );
-        }
+    std::list< std::pair<skill_id, int> > skillList;
+
+    if( skill_used != skill_id( "none" ) ) {
```

Comment:
```suggestion
    if( !skill_used.is_null() ) {
```
Please avoid magic values.